bluesky/tests/test_run_engine.py

```diff
@@ -571,34 +571,41 @@ def simple_plan(motor):
     assert motor.position == 1024
 
 
+@pytest.mark.skipif(
+    os.environ.get('TRAVIS', None) == 'true' and sys.platform == 'darwin',
+    reason=("The file-descriptor wake up based signal handling "
+            "does not work on travis on OSX"))
 def test_sigint_three_hits(RE, hw):
     import time
     motor = hw.motor
-    motor.delay = 1
+    motor.delay = .5
 
     pid = os.getpid()
 
-    def sim_kill(n=1):
+    def sim_kill(n):
         for j in range(n):
-            time.sleep(.02)
+            time.sleep(.05)
             os.kill(pid, signal.SIGINT)
 
     lp = RE.loop
     motor.loop = lp
-    start = time.monotonic()
-    threading.Timer(.05, sim_kill, (3,)).start()
-    threading.Timer(.1, sim_kill, (3,)).start()
-    threading.Timer(.2, sim_kill, (3,)).start()
+
+    def self_sig_int_plan():
+        threading.Timer(.05, sim_kill, (3,)).start()
+        yield from abs_set(motor, 1, wait=True)
 
     start_time = ttime.time()
     with pytest.raises(RunEngineInterrupted):
-        RE(finalize_wrapper(abs_set(motor, 1, wait=True),
+        RE(finalize_wrapper(self_sig_int_plan(),
                             abs_set(motor, 0, wait=True)))
     end_time = ttime.time()
-    assert end_time - start_time < 0.2  # not enough time for motor to cleanup
+    # not enough time for motor to cleanup, but long enough to start
+    assert 0.05 < end_time - start_time < 0.2
     RE.abort()  # now cleanup
+
     done_cleanup_time = ttime.time()
-    assert done_cleanup_time - end_time > 0.3
+    # this should be 0.5 (the motor.delay) above, leave sloppy for CI
+    assert 0.6 > done_cleanup_time - end_time > 0.3
```

Comment:
```suggestion
    assert 0.3 < done_cleanup_time - end_time < 0.6
```

Does this make anyone else's head hurt? :(